### PR TITLE
Make http2 interop tests always pass, and instead give a report

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
This change makes http2 interop tests always pass.  In a subsequent PR, I will be updating the report generation to output a percent success bar, to show how compliant servers are.  This nicely solves the cloud to prod tests, the Go tests, and any other tests added for servers that are not yet compliant.

### CODE REVIEW

**Comment by jtattermusch** [comment]
The interop tests have failed.

The problem I have with only showing a percent success bar is that unless there is some condition under which the tests will show up as red, the tests would require someone to continously look at their results. Tests that never fail seem to have no value to me.

**Comment by carl-mastrangelo** [comment]
The test failure is due to grpc-java, not this PR.  @ejona86 might know more since I believe he is doing a version bump.

As for always passing:  I think this would be better than always failing, which is what the alternative would be.  The problem is that if I find a new place where the spec isn't being followed, I make issues for the team that needs to fix them.  Then, I have to wait until each team fixes each issue before I can submit the test as part of the suite.  Until each language is thoroughly tested and stable, I don't think its feasible to make http2 interop tests blocking. 

Consider a likely case:  a low priority, pedantic bug is found in grpc.  The cost of fixing the bug is relatively high compared to value gained of having a correct implementation.  (lots of other things like feature work probably take priority).  We still should fix the bug, but the change that includes the test can't be added because it would make the tests red.   

I also considered making these tests threshold based, something like 95% success rate, or at most 3 failures, or something like that.  The risk there is that a low priority bug that wont be fixed soon might block the addition of a new test suite.  If any project was teetering on the failure threshold, no new tests could be added for fear of making the tests suddenly fail.  Obviously also bad!

Therefore, I believe that making these always pass is the right thing until they are more mature.

**Comment by ctiller** [comment]
We need to be able to both expand the test suite and loudly fail
regressions.

I'd suggested adding a flag (or some such) to the test suite so that we can
run one suite on presubmits -- which should always be green -- and another
on master (or elsewhere) that's non-blocking where we can expand at will
and file bugs on failures. Once the bugs clear, those tests could be moved
to the presubmit suite (and so stay green).

**Comment by carl-mastrangelo** [comment]
@ctiller : is run_interop_tests.py called from multiple places?  Also I am not sure what presubmit means specifically in regards to this.  Master is a branch and presubmit is an invocation of tests before commit right?

**Comment by ctiller** [comment]
Master and presubmit are two independent Jenkins jobs that run on different
triggers with different command lines.

**Comment by ejona86** [comment]
I think there should be a way to ignore tests for particular servers, and that should probably apply to both master and PRs. Tests we ignore we should have an issue open for, but it shouldn't turn any of the builds red. If it turns the build red, then I would expect we would perpetually be red which means everyone will ignore it.

The need to disable tests for certain cases also exists for the interop tests, and the "file issue, disable test" is the same approach we take there.

**Comment by ejona86** [comment]
@carl-mastrangelo, I don't see any java failure.

**Comment by carl-mastrangelo** [comment]
@jtattermusch PTAL.  I made it so that tests that are prefixed "TestSoon" don't block the overall suite failing.

File: tools/run_tests/run_interop_tests.py [hunk 1]
```diff
@@ -661,9 +661,6 @@ def build_interop_image_jobspec(language, tag=None):
           
     if args.http2_interop:
       for test_case in _HTTP2_TEST_CASES:
-        if server_name == "go":
```
Comment (jtattermusch) [inline tools/run_tests/run_interop_tests.py]: Looks like this change has introduces these failures:
tests.cloud_to_cloud:http2:go_server:tls
tests.cloud_to_cloud:http2:go_server:framing

**Comment by jtattermusch** [comment]
The ruby interop test failure is unrelated.

**Pull request merged**